src/ReachSets/discretize.jl

```diff
@@ -1,97 +1,185 @@
+of the matrix exponential over vectors when needed. Use the option
+`expm_method="lazy"` for this.
 """
-function discretize(cont_sys::InitialValueProblem{<:AbstractContinuousSystem},
+function discretize(𝑆::InitialValueProblem{<:AbstractContinuousSystem},
                     δ::Float64;
                     approx_model::String="forward",
-                    pade_expm::Bool=false,
-                    lazy_expm::Bool=false,
-                    lazy_sih::Bool=true)::InitialValueProblem{<:AbstractDiscreteSystem}
+                    expm_method::String="base",
+                    sih_method::String="lazy")
+
+    if !isaffine(S)
+        throw(ArgumentError("`discretize` is only implemented for affine ODEs"))
+    end
 
     if approx_model in ["forward", "backward"]
-        return discr_bloat_interpolation(cont_sys, δ, approx_model, pade_expm,
-                                         lazy_expm, lazy_sih)
+        return discr_bloat_interpolation(𝑆, δ, approx_model, expm_method, sih_method)
     elseif approx_model == "firstorder"
-        return discr_bloat_firstorder(cont_sys, δ)
+        return _discretize_first_order(𝑆, δ, expm_method)
     elseif approx_model == "nobloating"
-        return discr_no_bloat(cont_sys, δ, pade_expm, lazy_expm)
+        return discr_no_bloat(𝑆, δ, expm_method)
     else
-        error("The approximation model is invalid")
+        throw(ArgumentError("the approximation model $approx_model is unknown"))
+    end
+end
+
+function exp_Aδ(A::AbstractMatrix, δ::Float64, expm_method="base")
+    if expm_method == "base"
+        return expmat(Matrix(A*δ))
+    elseif expm_method == "lazy"
+        return SparseMatrixExp(A*δ)
+    elseif expm_method == "pade"
+        return padm(A*δ)
+    else
+       throw(ArgumentError("the exponentiation method $expm_method is unknown"))
     end
 end
 
 """
-    bloat_firstorder(cont_sys, δ)
+    _discretize_first_order(𝑆, δ, [p], [expm_method])
 
-Compute bloating factors using first order approximation.
+Apply a first order approximation model to `S` obtaining a discrete initial value problem.
 
-## Input
+### Input
 
-- `cont_sys` -- a continuous affine system
-- `δ`        -- step size
+- `𝑆` -- initial value problem for a continuous affine ODE with nondeterministic inputs
+- `δ` -- step size
+- `p` -- (optional, default: `Inf`) parameter in the considered norm
 
-## Notes
+### Output
 
-In this algorithm, the infinity norm is used.
-See also: `discr_bloat_interpolation` for more accurate (less conservative)
-bounds.
+The initial value problem for a discrete system.
 
-## Algorithm
+### Algorithm
+
+Let us define some notation. Let ``𝑆 : x' = Ax(t) + u(t)``,
+``x(0) ∈ \\mathcal{X}_0``, ``u(t) ∈ U`` be the given continuous affine ODE `𝑆`,
+where `U` is the set of nondeterministic inputs and ``\\mathcal{X}_0`` is the set
+of initial states.
+
+Let ``R_{X0} = \\max_{x∈ X0} ||x||`` and `D_{X0} = \\max_{x, y ∈ X0} ||x-y||``
```

Comment:
```suggestion
Let ``R_{X0} = \\max_{x ∈ \\mathcal{X}_0} ‖x‖`` and `D_{X0} = \\max_{x, y ∈ \\mathcal{X}_0} ‖x-y‖``
```

Reply:
I used `\\Vert` for `||`. If you agree, this needs to be fixed at other places as well.

Reply:
Yes, thanks. I hadn't yet polished the math typesetting.